javalib/src/main/scala/java/lang/IEEE754Helpers.scala

```diff
@@ -0,0 +1,68 @@
+package java.lang
+
+import scalanative.unsafe._
+import scalanative.libc.errno
+
+import scala.annotation.{switch, tailrec}
+import scalanative.posix.errno.ERANGE
+
+private[java] object IEEE754Helpers {
+
+  // Java parseDouble() and parseFloat() allow characters at & after
+  // after where C strtod() or strtof() stops. Continuous trailing whitespace
+  // with or without an initial 'D', 'd', 'F', 'f' size indicator
+  // is allowed.
+  // The whitespace can include tabs ('\t') and even newlines! ('\n').
+  // Perhaps even Unicode....
+  //
+  // The Java trim method is used to ensure that Java rules are followed.
+  // It might not be the most memory & runtime efficient, but if execution
+  // is in this routine, it is already on a slow path.
+
+  @tailrec
+  def vetIEEE754Tail(tail: String): Boolean = {
+
+    if (tail.length <= 0) {
+      true
+    } else {
+      val rest = tail.tail.trim
+      (tail(0): @scala.annotation.switch) match {
+        case 'D' | 'd' => vetIEEE754Tail(rest)
+        case 'F' | 'f' => vetIEEE754Tail(rest)
+        case _         => (tail.trim.length == 0)
+      }
+    }
+  }
+
+  def parseIEEE754[T](s: String, f: (CString, Ptr[CString]) => T): T = {
+    Zone { implicit z =>
+      val cstr = toCString(s)
+      val end  = stackalloc[CString]
+
+      errno.errno = 0
+      var res = f(cstr, end)
+
+      def exceptionMsg = s"For input string \042${s}\042"
```

Comment:
```suggestion
      def exceptionMsg = s"For input string \"$s\""
```

Reply:
\" not working in interpolated strings is a problem still a problem in many scala versions. It is a problem
in the Scala 2.11 currently used by SN and in the pending Scala 2.12.  I believe that it is unfixed in 
Scala 2.13.2. It may get fixed in 2.13.3 (or .n).

To reduce the donkeying around, I re-wrote the line to achieve the same intent but not rely upon
string interpolation.   I also added an informative section documenting the bug I was working around.

I wanted to steer far clear of octal constants (\042), which work, and Unicode constants (\u0022), which
fail because they are translated too early.